central/cloudsources/manager/manager_test.go

```diff
@@ -0,0 +1,123 @@
+package manager
+
+import (
+	"context"
+	"testing"
+
+	clusterDS "github.com/stackrox/rox/central/cluster/datastore"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/cloudsources/discoveredclusters"
+	"github.com/stackrox/rox/pkg/set"
+	"github.com/stretchr/testify/assert"
+)
+
+type mockClusterStore struct {
+	clusters []*storage.Cluster
+
+	clusterDS.DataStore
+}
+
+func (m *mockClusterStore) WalkClusters(_ context.Context, fn func(obj *storage.Cluster) error) error {
+	for _, cluster := range m.clusters {
+		_ = fn(cluster)
+	}
+	return nil
+}
+
+func TestMatchDiscoveredClusters(t *testing.T) {
+	clusterStore := &mockClusterStore{clusters: []*storage.Cluster{
+		createCluster(
+			"MC_testing_test-cluster-1_eastus",
+			"test-cluster-1",
+			storage.ClusterMetadata_AKS,
+		),
+		createCluster("something-else", "something-else", storage.ClusterMetadata_ROSA),
```

Comment:
I would suggest to use the real schema for `id`. I.e. uuid4 for openshift, arn for EKS.